### PR TITLE
 rhn_channel: fix broken import/Py3 compat and add unit tests

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
Fix [broken import](https://github.com/ansible/community/wiki/Testing%3A-progress-tracker#fixing-broken-imports) in `rhn_channel` module.

Other updates included:
- update `DOCUMENTATION` metadata
- remove commented statements
- fix `exit_json` usage

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
packaging/os/rhn_channel.py

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel 7bc85f9b44) last updated 2017/07/14 11:19:12 (GMT +200)
```

### CODE REVIEW

**Comment by vincentvdk** [comment]
ship it!

**Comment by pilou-** [comment]
@vincentvdk the bot recognizes [`shipit`](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#cmd-shipit) but not `ship it!`, could you please update your comment :) ?

File: lib/ansible/modules/packaging/os/rhn_channel.py [hunk 1]
```diff
@@ -146,8 +130,8 @@ def main():
     user = module.params['user']
     password = module.params['password']
 
-    #initialize connection
-    client = xmlrpclib.Server(saturl, verbose=0)
+    # initialize connection
+    client = xmlrpc_client(saturl)
```
Comment (mkrizek) [inline lib/ansible/modules/packaging/os/rhn_channel.py]: Same as in https://github.com/ansible/ansible/pull/26878, `xmlrpc_client.ServerProxy(saturl)`?

**Review by mkrizek — COMMENTED** [review]
This needs to run through shippable again.

**Comment by samdoran** [comment]
needs_revision

**Comment by pilou-** [comment]
Pull request rebased and updated:
- unit tests added
- fix 'module' object is not callable

**Review by mkrizek — APPROVED** [review]
LGTM :+1:

**Comment by abadger** [comment]
Merged to devel for the 2.4.0 release.

**Pull request merged**